### PR TITLE
fix(channel): pass claude system prompt via --append-system-prompt-file to avoid OS argv-length limits (spawn ENAMETOOLONG)

### LINKED ISSUE #554: [Bug] claude channel worker dies silently with spawn ENAMETOOLONG on Windows when injected context exceeds the 32K command-line limit
# [Bug] channel worker (claude provider) dies silently with `spawn ENAMETOOLONG` on Windows when injected context exceeds the 32K command-line limit

## Summary

On Windows, `trellis channel spawn --provider claude` with a large injected context (`--file` / `--jsonl`, e.g. a 244KB document) produces a worker that **dies instantly and silently** — no `reviewer.log` is ever created, no error event reaches the channel, and `trellis channel list` just shows the worker as gone. Running the supervisor in the foreground reveals the actual failure:

```
Supervisor error: spawn ENAMETOOLONG
```

## Environment

- OS: Windows 11 (also affects Linux at higher thresholds — see "Root cause")
- trellis: `@mindfoldhq/trellis@0.7.0-beta.3`
- claude CLI: `2.1.224` (Claude Code, WinGet install, native `claude.exe`)
- The claude CLI itself works fine in the same shell (`claude -p "say ok"` succeeds), and the exact argument set used by the adapter succeeds when the system prompt is small.

## Root cause

`dist/commands/channel/adapters/claude.js` → `buildClaudeArgs()` passes the entire system prompt — which includes all `--file` / `--jsonl` injected content — as a **command-line argument**:

```js
if (opts.systemPrompt?.trim()) {
    args.push("--append-system-prompt", opts.systemPrompt);
}
```

Then `supervisor.js` does `spawn(resolvedProvider.command, [...args])`.

- **Windows**: `CreateProcess` caps the command line at **32,767 characters**. A 244KB injected document → `spawn ENAMETOOLONG`.
- **Linux**: `MAX_ARG_STRLEN` (128KB per arg) would fail similarly for large contexts, just at a higher threshold.

## Why the failure is completely silent

Two compounding factors:

1. The dispatcher (`spawn.js`) launches the supervisor with `stdio: "ignore"`, so the supervisor's `Supervisor error: spawn ENAMETOOLONG` message on stderr is discarded.
2. The supervisor's `reviewer.log` is created lazily by `fs.createWriteStream`; when the process exits immediately after the spawn error, the async open never completes and the log file never appears.

Result: the only observable symptom is "worker vanished, no logs" — which looks like an environment/sandbox problem and is easy to misdiagnose (we initially blamed the IDE sandbox).

## Reproduction

```bash
# 1. Create a channel and spawn a claude worker with a large file injected
trellis channel create repro-enametoolong --by implementer
trellis channel spawn repro-enametoolong \
  --agent check --provider claude --as reviewer \
  --file ./docs/some-244kb-doc.md \
  --cwd "$PWD" --timeout 2m
# → spawn reports a pid; worker dies instantly; <worker>.log never created

# 2. Run the supervisor in the foreground to see the real error
TRELLIS_CHANNEL_PROJECT=<bucket> node <trellis-dist>/cli/index.js \
  channel __supervisor repro-enametoolong reviewer <config-path>
# → Supervisor error: spawn ENAMETOOLONG
```

With the same command but **no** large `--file`, the claude worker starts and runs normally — confirming the size of the command line is the only variable.

## Suggested fixes

1. **Preferred**: use the claude CLI's native `--append-system-prompt-file <path>` option (supported since claude CLI 2.x — `claude --help` shows `--system-prompt[-file], --append-system-prompt[-file]`). Write the system prompt to a file next to the worker artifacts (e.g. `<channel>/<worker>.system-prompt.md`) and pass only the path on the command line. This removes the length limit entirely.
2. **Defense in depth**: before `spawn()`, if the total length of the command line exceeds a safe threshold (~30K chars), fail fast with a clear error (and write it to the channel events + the worker log) instead of relying on `ENAMETOOLONG`.
3. **Observability**: don't launch the supervisor with fully discarded stdio — at minimum tee supervisor stderr into the worker log so startup crashes are diagnosable.

Happy to submit a PR for #1 (+#2) if the approach sounds good.

## Workaround (for anyone hitting this today)

Keep the total `--file` / `--jsonl` injected content under ~30KB; put paths to large documents in the first channel message instead and let the worker read them with its own tools. The codex provider is unaffected (its adapter passes the system prompt over stdin via `app-server` JSON-RPC).

### PR DESCRIPTION
**Pull request by lingbohome**
Fixes #554

## Problem

On Windows, `trellis channel spawn --provider claude` with a large injected context (`--file` / `--jsonl`, e.g. a 244KB document) dies instantly and silently: no `reviewer.log` is ever created, no error event reaches the channel. Running the supervisor in the foreground shows the real failure:

```
Supervisor error: spawn ENAMETOOLONG
```

Root cause: the claude adapter inlines the whole system prompt (channel protocol prefix + agent body + all injected context) as an `--append-system-prompt` **command-line argument**. Windows `CreateProcess` caps the command line at 32,767 chars; Linux `MAX_ARG_STRLEN` caps a single arg at 128KiB. The failure is invisible because the dispatcher launches the supervisor with `stdio: "ignore"` and the supervisor dies before its lazy `createWriteStream` log file materializes.

## Fix

Pass the system prompt via claude CLI's native file flag instead of inlining it on the command line:

- **supervisor.ts** — persist a non-empty `config.systemPrompt` to `<worker>.system-prompt.md` in the channel dir (same place as the other worker artifacts, already writable) and expose the path as `view.systemPromptFile`.
- **adapters/index.ts** — `SupervisorView` gains an optional `systemPromptFile`; the claude adapter forwards it.
- **adapters/claude.ts** — `buildClaudeArgs` prefers `--append-system-prompt-file <path>` when `systemPromptFile` is set, and falls back to the old inline `--append-system-prompt` otherwise (backwards compatible; codex unaffected — its prompt goes over stdin JSON-RPC).

The generated claude command line is now constant-size regardless of how much context is injected, removing the OS limit on both platforms.

## Test plan

- New `packages/cli/test/commands/channel-claude-adapter.test.ts` covers: inline fallback when no file is given, blank-prompt skip, and the file branch asserting a 300K-char prompt no longer appears on the command line.
- Manually verified on Windows against the repro from #554: a claude worker spawned with a 244KB `--file` now starts, processes messages, and shuts down on timeout (previously: instant silent ENAMETOOLONG).

## Summary by CodeRabbit

* **New Features**
  * Claude now supports loading system prompts from files.
  * Inline prompts remain supported for shorter prompts.
  * Large prompts are handled without exceeding command-line argument limits.

* **Bug Fixes**
  * Blank system prompts no longer add unnecessary command-line options.
  * Temporary system prompt files are removed during worker cleanup.

* **Tests**
  * Added coverage for inline, file-based, blank, and oversized system prompts.

### CODE REVIEW

**Comment by lingbohome** [comment]
Addressed the compatibility concern with a threshold fallback — the second option from the review (preserve existing inline behavior for older installations):

- **New:** `INLINE_SYSTEM_PROMPT_MAX_CHARS = 8_000` + `shouldUseSystemPromptFile()` (exported from `adapters/claude.ts`). Prompts at or below 8,000 chars stay on the inline `--append-system-prompt` argv flag, so Claude Code installs older than v2.0.34 (which lack `--append-system-prompt-file`) keep working exactly as before. Only oversized prompts are persisted to `<worker>.system-prompt.md` and passed via the file flag.
- **Why 8,000:** the failure mode being fixed is the Windows CreateProcess 32,767-char command-line cap; 8,000 leaves ~4x headroom for the fixed flags plus Windows quoting/escaping expansion, while still covering every prompt that could safely be inlined in practice.
- **Tests:** added `shouldUseSystemPromptFile` boundary cases (blank → inline, exactly 8,000 → inline, 8,001 → file, 175KB → file) alongside the existing precedence tests.
- **Verified locally (end-to-end, Windows):** small agent prompt spawns with the inline flag and no prompt file; a 17KB prompt (12KB `--file` injection + agent body) spawns with `--append-system-prompt-file`; and inlining a 175KB prompt still reproduces the original `ENAMETOOLONG`, confirming the threshold is what keeps both worlds working.

File: packages/cli/src/commands/channel/supervisor.ts [hunk 1]
```diff
@@ -1,586 +1,609 @@
...
+  let systemPromptFile: string | undefined;
+  if (shouldUseSystemPromptFile(config.systemPrompt)) {
+    systemPromptFile = workerFile(
+      channelName,
+      workerName,
+      "system-prompt.md",
+      project,
+    );
+    fs.writeFileSync(systemPromptFile, config.systemPrompt);
+  }
```
[the remark this answers is not in the document]
Comment (lingbohome) [reply packages/cli/src/commands/channel/supervisor.ts]: Fixed — and thank you, this surfaced a second gap worth calling out: the proposed diff alone is not sufficient on Windows.

`trellis channel kill` sends SIGKILL to the supervisor, and on Windows that terminates the process without running any of its handlers — so the supervisor-side `cleanup()` you pointed at never executes on the external-kill path. The CLI kill command has its own artifact list (`cleanupFiles()` in `kill.ts`), which is what actually runs there.

Both lists now include the file:

- `supervisor.ts` `cleanup()` — covers worker-exit paths (spawn failure, natural exit, supervisor-driven timeout kill)
- `kill.ts` `cleanupFiles()` — covers the external CLI kill path (the fallback when the supervisor cannot clean up after itself)

Removed unconditionally in both, so a later inline-prompt run also clears a stale prompt file from a previous oversized run, exactly as you noted.

Verified on Windows for both exit paths with a 17KB prompt (12KB `--file` injection + agent body):

- CLI kill → prompt file removed, `log` kept (forensic), `session-id`/`thread-id` keep-list untouched
- supervisor timeout exit → prompt file removed via the supervisor's own cleanup

**Comment by taosu0216** [comment]
Merged — thanks for tracking this down. Silent worker death with no log line is about the worst failure shape there is, and your diagnosis of *why* it was silent (dispatcher using `stdio: "ignore"`, plus the worker log being created lazily so the process died before the stream existed) was more useful than the fix itself for understanding what else might be hiding.

One thing I did before merging, so you are not surprised by the extra commit on your branch: four files had been saved with CRLF, which git recorded as a full rewrite — 2,159 changed lines for roughly 75 lines of real work, and three files that would have landed on `main` with different line endings than everything around them. I normalized them back to LF (`4809a7f6`, content untouched) rather than send you round again for something mechanical.

I have also added `* text=auto eol=lf` to `.gitattributes` on `main` (`bd454938`). The repo was already all-LF, so it changes nothing today — it just stops this from happening to the next contributor working on Windows. That gap was ours, not yours; you were the first to hit it.

## One limitation worth recording

The 8,000-character threshold keeps old Claude Code versions working, since `--append-system-prompt-file` only exists in 2.0.34+. But someone on an older CLI who injects more than 8,000 characters still fails — and fails the same silent way, because the `stdio: "ignore"` path is unchanged. Narrower than before (it used to fail for everyone at 32K), but not gone.

If you have appetite for a follow-up, the higher-value one is probably making that failure loud rather than extending the version handling: surfacing the supervisor's stderr so a spawn failure produces *something* in the channel. That would have turned this bug from a silent death into a one-line error, and it would cover the failures we have not thought of yet. Entirely optional — this PR stands on its own.
